### PR TITLE
Global GAPIC refresh

### PR DESCRIPTION
**Pull request by geigerj**
This updates all GAPICs to support `googleauth` 0.6.1. It additionally makes minor updates to the manual surfaces where there have been changes to GAPIC method signatures, and updates other `google-cloud-ruby` dependencies for googleauth support.

### CODE REVIEW

File: google-cloud-bigtable/Rakefile [hunk 1]
```diff
@@ -122,7 +122,7 @@ end
 
 namespace :ci do
   desc "Run the CI build, with acceptance tests."
-  task :acceptance do
+  task :smoke_test do
```
Comment (blowmage) [inline google-cloud-bigtable/Rakefile]: The build needs an `acceptance` task to exist. Why is this renamed?

Comment (geigerj) [reply google-cloud-bigtable/Rakefile]: This is a mistake. I missed a replacement when updating the code generator templates. (This PR is not ready for review yet.)

Comment (blowmage) [reply google-cloud-bigtable/Rakefile]: Gotcha. Understood on WIP. I just like to be familiar with upcoming changes and watch the progress. Looks great so far!

**Comment by quartzmo** [comment]
99 Datastore doctest errors failing on:

```sh
ArgumentError: mocked method :run_query expects 4 arguments, got 3

    /home/ubuntu/google-cloud-ruby/google-cloud-datastore/lib/google/cloud/datastore/service.rb:103:in `block in run_query'

    /home/ubuntu/google-cloud-ruby/google-cloud-datastore/lib/google/cloud/datastore/service.rb:166:in `execute'

    /home/ubuntu/google-cloud-ruby/google-cloud-datastore/lib/google/cloud/datastore/service.rb:102:in `run_query'

    /home/ubuntu/google-cloud-ruby/google-cloud-datastore/lib/google/cloud/datastore/dataset.rb:458:in `run'

    /opt/circleci/.rvm/gems/ruby-2.4.0/gems/yard-doctest-0.1.8/lib/yard/doctest/example.rb:79:in `context'

    /home/ubuntu/google-cloud-ruby/google-cloud-datastore/lib/google/cloud/datastore/dataset/query_results.rb:337

    /opt/circleci/.rvm/gems/ruby-2.4.0/gems/yard-doctest-0.1.8/lib/yard/doctest/example.rb:56:in `eval'

    /opt/circleci/.rvm/gems/ruby-2.4.0/gems/yard-doctest-0.1.8/lib/yard/doctest/example.rb:56:in `evaluate_example'

    /opt/circleci/.rvm/gems/ruby-2.4.0/gems/yard-doctest-0.1.8/lib/yard/doctest/example.rb:43:in `block (4 levels) in generate'

    /opt/circleci/.rvm/gems/ruby-2.4.0/gems/yard-doctest-0.1.8/lib/yard/doctest/example.rb:40:in `each'

    /opt/circleci/.rvm/gems/ruby-2.4.0/gems/yard-doctest-0.1.8/lib/yard/doctest/example.rb:40:in `block (3 levels) in generate'
```

**Comment by quartzmo** [comment]
This also occurs in Datastore `rake test` (which would run after `doctest`), with 44 similar errors:

` 44) Error:
Google::Cloud::Datastore::Dataset::QueryResults::mock_datastore#test_0001_has more_results not_finished:
ArgumentError: mocked method :run_query expects 4 arguments, got 3`

**Comment by geigerj** [comment]
@quartzmo Whoops. Looks like I still have some mocks to update. Will try to get a fix this afternoon.

**Comment by quartzmo** [comment]
I also get this rubocop failure in `google-cloud-language`:

```sh
Offenses:

lib/google/cloud/language/service.rb:90:36: C: Align the elements of a hash literal if they span more than one line.
                                   options: default_options
                                   ^^^^^^^^^^^^^^^^^^^^^^^^
lib/google/cloud/language/service.rb:97:38: C: Align the elements of a hash literal if they span more than one line.
                                     options: default_options
                                     ^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by quartzmo** [comment]
As well as `doctest` and `test` errors in `google-cloud-language`. Here is a test error:

```sh
 89) Error:
Google::Cloud::Language::Project::full_text_annotation::mock_language::document GSC object#test_0001_runs full annotation with document object using empty options:
ArgumentError: mocked method :annotate_text expects 4 arguments, got 3
```

**Comment by quartzmo** [comment]
`doctest` errors in `google-cloud-pubsub`:

```sh
 15) Error:
Google::Cloud::Pubsub::Snapshot#delete#test_0001_:
Google::Protobuf::ParseError: Error occurred during parsing: No such field: expiration_time
```

**Comment by quartzmo** [comment]
Now I'm getting these  `doctest` errors in `google-cloud-debugger`:

```sh
 73) Error:
Google::Cloud::Debugger::Project::mock_debugger#test_0001_knows the project identifier:
Errno::EHOSTDOWN: Failed to open TCP connection to 169.254.169.254:80 (Host is down - connect(2) for "169.254.169.254" port 80)
```

**Comment by quartzmo** [comment]
Other than the packages mentioned above, `bundle exec rake ci[yes]` in the top-level directory passes for me. Hope this helps. Excited to get this PR merged.

**Comment by quartzmo** [comment]
Still failing on Datastore `doctest`. I get the same errors locally.

**Comment by geigerj** [comment]
@quartzmo Yep, still working on doctest. A lot of mocks need to be updated. I'll comment when ready for another review pass -- thanks for the detailed comments.

**Comment by geigerj** [comment]
@quartzmo Woohoo, build passes now. I think the Debugger errors you observed were transient network issues. Everything else seems to have been resolved by updating test/doctest mocks.

File: google-cloud-core/Gemfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ gemspec
 
 gem "rake", "~> 11.0"
 gem "google-api-client", "~> 0.14.0"
```
Comment (blowmage) [inline google-cloud-core/Gemfile]: The google-api-client version needs to be updated to `~> 0.16.0` to match the version change made everywhere else.

File: google-cloud-bigquery/google-cloud-bigquery.gemspec [hunk 3]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |gem|
   gem.required_ruby_version = ">= 2.0.0"
 
   gem.add_dependency "google-cloud-core", "~> 1.0"
-  gem.add_dependency "google-api-client", "~> 0.14.0"
+  gem.add_dependency "google-api-client", "~> 0.16.0"
```
Comment (blowmage) [inline google-cloud-bigquery/google-cloud-bigquery.gemspec]: Are Bigquery Credentials changing to inherit from `Google::Auth::Credentials`? If so, then a dependency on `googleauth ~> 0.6.1` is needed here as well.

File: google-cloud-monitoring/Rakefile [hunk 4]
```diff
@@ -114,9 +143,19 @@ task :ci do
...
+    Rake::Task["ci"].invoke
+    header "google-cloud-monitoring smoke_test", "*"
+    sh "bundle exec rake smoke_test -v"
+  end
+end
+
 namespace :ci do
   desc "Run the CI build, with acceptance tests."
-  task :acceptance do
+  task :acccpetance do
```
Comment (blowmage) [inline google-cloud-monitoring/Rakefile]: This task name should not be changed.

File: google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb [hunk 5]
```diff
@@ -121,19 +121,7 @@ class PolicyDelta; end
       #     A single identity requesting access for a Cloud Platform resource.
       #     Follows the same format of Binding.members.
       #     Required
-      class BindingDelta
```
Comment (blowmage) [inline google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb]: Is this removal still expected? See https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1727#discussion_r139265901

Comment (geigerj) [reply google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb]: I suspect this is a regression, but would like to decouple it from this refresh. I filed https://github.com/googleapis/toolkit/issues/1678 to investigate.

File: google-cloud-monitoring/Rakefile (hunk 4 above)
Comment (geigerj) [reply google-cloud-monitoring/Rakefile]: Done.

File: google-cloud-bigquery/google-cloud-bigquery.gemspec (hunk 3 above)
Comment (geigerj) [reply google-cloud-bigquery/google-cloud-bigquery.gemspec]: Not until https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/1729 is done. Even after that issue is resolved, I'd expect the dependency to be pull in transitively via `google-cloud-core` rather than in the individual gemspecs for the gems with manual layers.

File: google-cloud-datastore/lib/google/cloud/datastore/v1/datastore_client.rb [hunk 6]
```diff
@@ -93,14 +103,38 @@ def initialize \
             require "google/gax/grpc"
             require "google/datastore/v1/datastore_services_pb"
 
+            if channel || chan_creds || updater_proc
+              warn "The `channel`, `chan_creds`, and `updater_proc` parameters will be removed " \
+                "on 2017/09/08"
```
Comment (quartzmo) [inline google-cloud-datastore/lib/google/cloud/datastore/v1/datastore_client.rb]: I saw this output while running tests and it made me wonder about the date, which is already past... Does the date need to be updated?

Comment (geigerj) [reply google-cloud-datastore/lib/google/cloud/datastore/v1/datastore_client.rb]: I'm going to remove these parameters from the code generator. It should propagate out to the GAPIC clients on the next refresh.

File: google-cloud-monitoring/Rakefile [hunk 7]
```diff
@@ -21,17 +19,53 @@ RuboCop::RakeTask.new
...
+      track_files "lib/**/*.rb"
+      add_filter "test/"
+    end
+
+    Rake::Task["test"].invoke
+  end
+end
+
+desc "Runs the smoke tests."
+task :smoke_test do
```
Comment (quartzmo) [inline google-cloud-monitoring/Rakefile]: Should `smoke_test` have been renamed `acceptance` here (monitoring)?

File: google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb (hunk 5 above)
Comment (blowmage) [reply google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb]: Should this block approval and merging of this PR?

Comment (geigerj) [reply google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb]: I manually reverted this. (I grepped through the diff, and this seems to be the only occurrence where a module within a doc file has gone missing.)

File: google-cloud-monitoring/Rakefile (hunk 7 above)
Comment (geigerj) [reply google-cloud-monitoring/Rakefile]: The `smoke_test`s (which should start being added soon...) are a subset of `acceptance` tests, so I'm keeping the tasks separate. For GAPIC-only releases, the smoke test may be the only acceptance test, but for others, we might want a way to invoke manual system tests separately from autogen smoke tests.

Comment (blowmage) [reply google-cloud-monitoring/Rakefile]: I do worry about the smoke tests being in the `test` directory, because they make actual calls to the API. I worry that they may be invoked during the build, like during the coverage task which loads all files under `test`.

Comment (geigerj) [reply google-cloud-monitoring/Rakefile]: Acknowledged, but can we defer this discussion to when the smoke tests are added? No such smoke tests are included in this PR.

Comment (blowmage) [reply google-cloud-monitoring/Rakefile]: Yep, I'm not intending to request any changes here, just bringing it up.

File: google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb [hunk 8]
```diff
@@ -121,7 +121,19 @@ class PolicyDelta; end
       #     A single identity requesting access for a Cloud Platform resource.
       #     Follows the same format of Binding.members.
       #     Required
-      class BindingDelta; end
+      class BindingDelta
```
Comment (blowmage) [inline google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb]: I believe this change should also be applied to google-cloud-spanner/lib/google/cloud/spanner/admin/instance/v1/doc/google/iam/v1/policy.rb, yes?

Comment (geigerj) [reply google-cloud-spanner/lib/google/cloud/spanner/admin/database/v1/doc/google/iam/v1/policy.rb]: Done.
